nixos/modules/services/web-apps/pingvin-share.nix

```
@@ -0,0 +1,233 @@
+        };
+      };
+
+      nginx = {
+        enable = mkEnableOption (mdDoc ''
+          exposing Pingvin Share with a Nginx reverse proxy.
+        '');
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.groups = mkIf (cfg.group == "pingvin") {
+      pingvin = { };
+    };
+
+    users.users = mkIf (cfg.user == "pingvin") {
+      pingvin = {
+        group = cfg.group;
+        # Need to set the user home because the package writes to "$HOME/.npm"
+        home = cfg.settings.dataDir;
+        description = "Pingvin Share daemon user";
+        isSystemUser = true;
+      };
+    };
+
+    systemd.services.pingvin-share-backend = {
+      description = "Backend service of Pingvin Share, a self-hosted file sharing platform and an alternative for WeTransfer.";
+
+      wantedBy = [ "multi-user.target" "pingvin-share-frontend.service" ];
+      wants = [ "network-online.target" ];
+      before = [ "pingvin-share-frontend.service" ];
+      after = [ "network-online.target" ];
+
+      environment =
+        let
+          prisma-engines = pkgs.prisma-engines;
+        in
+        {
+          PRISMA_SCHEMA_ENGINE_BINARY = "${prisma-engines}/bin/schema-engine";
+          PRISMA_QUERY_ENGINE_BINARY = "${prisma-engines}/bin/query-engine";
+          PRISMA_QUERY_ENGINE_LIBRARY = "${prisma-engines}/lib/libquery_engine.node";
+          PRISMA_INTROSPECTION_ENGINE_BINARY = "${prisma-engines}/bin/introspection-engine";
+          PRISMA_FMT_BINARY = "${prisma-engines}/bin/prisma-fmt";
+
+          PORT = "${toString cfg.backend.port}";
+          DATABASE_URL = "file:${cfg.settings.dataDir}/pingvin-share.db?connection_limit=1";
+          DATA_DIRECTORY = "${cfg.settings.dataDir}";
+        };
+
+      path = with pkgs; [
+        cfg.backend.package
+
+        bash
+        openssl
+        nodejs
+        prisma-engines
+      ];
+
+      script = ''
+        ${pkgs.nodePackages.npm}/bin/npm run prod --prefix ${cfg.backend.package}
+      '';
+
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+
+        Type = "simple";
+        Restart = "on-failure";
+
+        StateDirectory = mkIf (cfg.settings.dataDir == "/var/lib/pingvin-share") "pingvin-share";
+      };
+    };
+
+    systemd.services.pingvin-share-frontend = {
+      description = "Frontend service of Pingvin Share, a self-hosted file sharing platform and an alternative for WeTransfer.";
+
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network-online.target" "pingvin-share-backend.service" ];
+      after = [ "network-online.target" "pingvin-share-backend.service" ];
+
+      environment = {
+        PORT = "${toString cfg.frontend.port}";
+        API_URL = if cfg.settings.hostname == null then "http://localhost:${toString cfg.backend.port}" else "${if cfg.settings.https then "https" else "http"}://${cfg.settings.hostname}";
+      };
+
+      path = with pkgs; [
+        cfg.frontend.package
+
+        bash
+      ];
+
```

```suggestion
      path = [ cfg.frontend.package ];
```
bash should be already in path or scripts patches.